components/engine_test/src/lib.rs

```diff
@@ -232,16 +235,17 @@ pub mod kv {
     }
 
     impl TestTabletFactoryV2 {
-        pub fn new(
-            root_path: &Path,
-            db_opt: DbOptions,
-            cf_opts: Vec<(&'static str, KvTestCfOptions)>,
-        ) -> Self {
+        pub fn new(root_path: &Path, db_opt: DbOptions) -> Self {
             Self {
-                inner: TestTabletFactory::new(root_path, db_opt, cf_opts),
+                inner: TestTabletFactory::new(root_path, db_opt),
                 registry: Arc::default(),
             }
         }
+
+        pub fn register_tablet(&self, region_id: u64, suffix: u64, tablet: KvTestEngine) {
```

Comment:
You should make `create_tablet` accept tablet ids, and use `open_tablet` in the test.

Reply:
This is just a helper function for multi_gc_engine_setup(), where engine_rocks::util::new_engine_opt is called to create tablet instead of open_tablet. All those new functions are for test mocking purpose only.

Reply:
I tried but only found TestTabletFactory::create_tablet() can't access structures like BlockCacheConfig, DbConfig etc due to the dependency issue.

Reply:
You need to manually create the compaction filter here (instead of reusing the `build_opt` in tikv crate).

Reply:
I don't see your proposal can make code more readable...

Reply:
It's not a problem of readability, it's not correct. The code should be able to build against different engine implementations, e.g. `cargo test --package tikv --features test-engines-panic --no-default-features`. To accomplish that you cannot associate a fixed type of engine with `TestTabletFactory`.

@SpadeA-Tang I run this locally and it appears you introduced a lot of changes that broke this build.

Reply:
@tabokie the problem should origins from @tonyxuqqi commit 03b44b5e219fb795b20ebb9367b66b2adf3800ac

Reply:
Fixing it in #13585.